File: td-frontend/refactor.py
import os
import re

directory = r'C:\Users\lisan\Desktop\Proyectos\TechDepot\td-frontend\src\pages\Admin'

button_pattern = re.compile(r'<Button\s+onClick=\{([^}]+)\}\s+variant="danger"\s+style=\{[^}]+\}\s*>\s*Eliminar\s*</Button>', re.DOTALL)
button_pattern2 = re.compile(r'<Button\s+variant="danger"\s+onClick=\{([^}]+)\}\s+style=\{[^}]+\}\s*>\s*Eliminar\s*</Button>', re.DOTALL)
button_pattern3 = re.compile(r'<Button\s+onClick=\{([^}]+)\}\s+variant="danger"\s*>\s*Eliminar\s*</Button>', re.DOTALL)


replacement_template = '''<button 
                                            type="button" 
                                            onClick={%s}
                                            style={{
                                                background: 'none',
                                                border: 'none',
                                                color: 'var(--danger-color)',
                                                cursor: 'pointer',
                                                fontWeight: 'bold',
                                                fontSize: '1.2rem',
                                                padding: '4px 8px'
                                            }}
                                            title="Eliminar"
                                        >
                                            ×
                                        </button>'''

for filename in os.listdir(directory):
    if filename.endswith('.tsx') and filename != 'PortManagerPage.tsx':
        filepath = os.path.join(directory, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        new_content = content
        
        def replacer(match):
            onclick = match.group(1)
            return replacement_template % onclick
            
        new_content = button_pattern.sub(replacer, new_content)
        new_content = button_pattern2.sub(replacer, new_content)
        new_content = button_pattern3.sub(replacer, new_content)
        
        if new_content != content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f'Updated {filename}')

File: td-frontend/refactor.mjs
import fs from 'fs';
import path from 'path';

const dir = 'C:\\Users\\lisan\\Desktop\\Proyectos\\TechDepot\\td-frontend\\src\\pages\\Admin';
const files = fs.readdirSync(dir).filter(f => f.endsWith('.tsx') && f !== 'PortManagerPage.tsx');

const regexes = [
    /<Button\s+onClick=\{([^}]+)\}\s+variant="danger"\s+style=\{[^}]+\}\s*>\s*Eliminar\s*<\/Button>/g,
    /<Button\s+variant="danger"\s+onClick=\{([^}]+)\}\s+style=\{[^}]+\}\s*>\s*Eliminar\s*<\/Button>/g,
    /<Button\s+onClick=\{([^}]+)\}\s+variant="danger"\s*>\s*Eliminar\s*<\/Button>/g
];

const replacement = `<button 
                                            type="button" 
                                            onClick={$1}
                                            style={{ background: 'none', border: 'none', color: 'var(--danger-color)', cursor: 'pointer', fontWeight: 'bold', fontSize: '1.2rem', padding: '4px 8px' }}
                                            title="Eliminar"
                                        >
                                            ×
                                        </button>`;

for (const file of files) {
    const filePath = path.join(dir, file);
    let content = fs.readFileSync(filePath, 'utf8');
    let newContent = content;
    
    for (const regex of regexes) {
        newContent = newContent.replace(regex, replacement);
    }
    
    if (newContent !== content) {
        fs.writeFileSync(filePath, newContent, 'utf8');
        console.log(`Updated ${file}`);
    }
}
